modules/lily-compiler/src/main/java/io/github/tomboyo/lily/compiler/icg/AstGenerator.java

```diff
@@ -27,7 +29,30 @@ public static Stream<Ast> evaluate(PackageName basePackage, OpenAPI openAPI) {
   }
 
   private Stream<Ast> evaluate(OpenAPI openAPI) {
-    return Stream.of(evaluateComponents(openAPI), evaluatePaths(openAPI)).flatMap(identity());
+    var table =
+        Stream.of(evaluateComponents(openAPI), evaluatePaths(openAPI))
+            .flatMap(identity())
+            /*
+             * Collect definitions into a FQN -> AST table. This gives us an
+             * opportunity to modify AST generated from schema encountered early based
+             * on schema encountered late. We have to do this to create sealed
+             * interfaces, for example, which reference and are referenced by their
+             * permitted members.
+             */
+            .collect(
+                Collectors.toMap(
+                    ast -> ast.name(),
+                    ast -> ast,
+                    (left, right) -> {
+                      if (left instanceof Modifier mod && right instanceof Definition def) {
+                        return mod.modify(def);
+                      } else if (left instanceof Definition def && right instanceof Modifier mod) {
+                        return mod.modify(def);
```

Comment:
Is there a way to rewrite this if statement to be more permissive since it does the same thing at the end? Or is it because of the variable creation in the if check that both are needed?